src/app/pages/dashboard/services/widget-resources.service.ts

```diff
@@ -196,6 +198,11 @@ export class WidgetResourcesService {
   }
 
   refreshSystemInfo(): void {
```

Comment:
Can you explain what's going on here? 
The confusion around a "get" method like `networkInterfaceLastHourStats` making changes to system store stands.

Reply:
the problem with the bug was that the WebUI was sending a request to the middleware with an outdated date stamp. To solve this, I called `systemInfoUpdated` to update it. But the subscription to `serverTime$` was running before this update, and the `take(1)` operator prevented it from running again after `serverTime$` was updated.

I set the `systemInfo` value to null so that `waitForSystemInfo` would wait for new data to arrive and the logic would work correctly.

Reply:
So if the problem is with `serverTime$`, can't we fix it somehow? Or why do we need `serverTime$` at all? Why can't we just take current timestamp and current timestamp minus one hour?

Reply:
Getting `serverTime$` from the middleware is necessary because of a time synchronization issue between the user's system and the server.
For example, time zones or DST may be different, and the request requires exact start and end times.

Reply:
But you use timestamps in `reporting.netdata_get_data`, so they are the same regardless of timezone or DST.

Reply:
What I meant was that `serverTime` may not match `new Date().getTime()` for various reasons. `serverTime` ensures that it is the current time for reporting.

Reply:
The only reason it may not match if either system has incorrect local time. I don't think we need to plan for this.